### PR TITLE
fix(package): update jest dependencies

### PR DESCRIPTION
**Pull request by SimenB**
(mostly to not rely on beta releases)

### CODE REVIEW

File: __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap [hunk 1]
```diff
@@ -5,32 +5,32 @@ Snapshot Diff:
...
--  "bar",
-   "baz",
-+  "quoz",
- ]
+  Array [
+    "foo",
+-   "bar",
+    "baz",
++   "quoz",
+  ]
```
Comment (thymikee) [inline __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: @SimenB do you have any ideas on why do we have extra space in the snapshots?

Comment (SimenB) [reply __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: I'm not sure. It's a diff for jest 21. I'll see if I can isolate it

Comment (SimenB) [reply __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: It's this: https://github.com/facebook/jest/commit/7e2c206364bb8d0b3e09b877dd658efc5c27458b#diff-cbe0ce708017e083e342adce28e3720d

/cc @pedrottimark

Comment (pedrottimark) [reply __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: Yes, that PR fixed a difference (pardon the pun :) between options:
* `expand: true` output a one-space “gutter” between -/+/space and content so content aligned with labels (see `can expand diff 1` below)
* `expand: false` did not, but now it does, so the snapshot will not change if you toggle the option (for a diff which has fewer lines than the context)

Comment (pedrottimark) [reply __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: See https://github.com/facebook/jest/pull/3429#issuecomment-326090900

File: __tests__/__snapshots__/snapshotDiff.test.js.snap [hunk 2]
```diff
@@ -52,31 +54,32 @@ exports[`can use contextLines with React components 1`] = `
...
+      some
+      some
+      some
+      some
+      not
++     so
+      very
+      long
+      script
+  "
```
Comment (pedrottimark) [inline __tests__/__snapshots__/snapshotDiff.test.js.snap]: @thymikee This change is from https://github.com/facebook/jest/pull/4520:
* `expand: false` option made sure that strings end with newline
* `expand: true` did not, and in this case it even trimmed the newline at end of expected and received strings!

**Comment by thymikee** [comment]
Thanks @SimenB @pedrottimark we'll need to mark it as breaking change then.

File: __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap (hunk 1 above)
Comment (thymikee) [reply __tests__/__snapshots__/getSnapshotDiffSerializer.test.js.snap]: You guys are awesome

**Comment by thymikee** [comment]
@SimenB ping, we have a CI now 😉 thanks for reminding me about it

File: package.json [hunk 3]
```diff
@@ -20,20 +20,20 @@
...
+    "pretty-format": "^22.0.6",
     "strip-ansi": "^4.0.0"
   },
   "devDependencies": {
     "babel-cli": "^6.24.1",
     "babel-preset-env": "^1.6.0",
     "babel-preset-flow": "^6.23.0",
     "babel-preset-react": "^6.24.1",
-    "eslint": "^4.3.0",
+    "eslint": "~4.13.0",
```
Comment (SimenB) [inline package.json]: I deleted the lockfile to update all deps within semver, and without also updating `eslint-config-callstack-io` we are hit by the babel-eslint bug.

**Comment by thymikee** [comment]
I'm cool with next release to be a major, happy to see #20 as a part of it. I'll try to squeeze some time tomorrow or so to do some work (would also be cool to add more plugins by default, e.g. ReactTestComponent, not only ReactElement)

File: package.json (hunk 3 above)
Comment (thymikee) [reply package.json]: Sure, good move

**Comment by pedrottimark** [comment]
@SimenB @thymikee For your info, when replace `diff` with `diff-sequences` in `jest-diff` it will also be a breaking change. If that seems like anytime soon, we might wait on it to avoid churn.

**Comment by thymikee** [comment]
@pedrottimark that's a valid point. We don't have a release schedule, so I'm ok with waiting a bit for the new release. WDYT @SimenB?

File: package.json (hunk 3 above)
Comment (thymikee) [reply package.json]: We can update the eslint config in the followup PR though

**Comment by SimenB** [comment]
It's annoying to see beta deps in my tree but there is no rush of course 🙂

**Comment by SimenB** [comment]
I cannot reproduce the failing node 6 test locally 🙁

**Comment by thymikee** [comment]
Looks like Travis being color blind again 🤷‍♂️

**Pull request merged**